shared/cmd/flags.go

```diff
@@ -107,6 +107,12 @@ var (
 			"would whitelist connections to peers on your local network only. The default " +
 			"is to accept all connections.",
 	}
+	// P2PEncoding defines the encoding format for p2p messages.
+	P2PEncoding = cli.IntFlag{
+		Name:  "p2p-encoding",
+		Usage: "The encoding format of messages sent over the wire. The default is 0, which represents ssz",
+		Value: 0,
+	}
```

Comment:
can we consider changing it from int to boolean something like ssz-compression ?

Reply:
@shayzluf we ideally want to support a number of encoding formats, ex: protobufs,json etc. This isnt solely for specifying snappy compression, so this is why I have the flag as an int

Reply:
sg